__tests__/app.test.js

```diff
@@ -40,124 +40,181 @@ describe("GET /api/topics", () => {
+        });
+      });
+      test("return array of comments is ordered with most recent first", () => {
+        return request(app).get("/api/articles/1/comments").then(({body}) => {
+          expect(body.comments.length).toBe(11);
+          expect(body.comments).toBeSortedBy("created_at", {descending: true});
+        });
+      });
+      test("return an empty array if the article exists but has no comments", () => {
+        return request(app).get("/api/articles/2/comments").expect(200).then(({body}) => {
+          expect(body.comments.length).toBe(0);
+        });
       });
     });
   });
-  describe("GET /api/articles/:article_id/comments", () => {
-    test("return an array of comment objects", () => {
-      return request(app).get("/api/articles/1/comments").expect(200).then(({body}) => {
-        expect(body.comments.length).toBe(11);
-        body.comments.forEach((comment) => {
-          expect(comment).toMatchObject({
+  describe("POST", () => {
+    describe("POST /api/articles/:article_id/comments", () => {
+      test("Respond with the newly created comment object", () => {
+        return request(app).post("/api/articles/1/comments").send({
+          username: "lurker",
+          body: "My first post!"
+        }).expect(201).then(({body}) => {
+          expect(body.comment).toMatchObject({
             article_id: 1,
             comment_id: expect.any(Number),
-            votes: expect.any(Number),
+            votes: 0,
             created_at: expect.any(String),
-            author: expect.any(String),
-            body: expect.any(String),
+            author: "lurker",
+            body: "My first post!",
           });
         });
       });
-    });
-    test("return 404 if article not found", () => {
-      return request(app).get("/api/articles/100000/comments").expect(404);
-    });
-    test("return 400 if article_id is not an integer", () => {
-      return request(app).get("/api/articles/first_article/comments").expect(400);
-    });
-    test("return the correct comments when provided with an article_id", () => {
-      return request(app).get("/api/articles/6/comments").then(({body}) => {
-        const result = [{
-          body: "This is a bad article name",
-          votes: 1,
-          author: "butter_bridge",
-          article_id: 6,
-          created_at: "2020-10-11T15:23:00.000Z"
-        }];
-        expect(body.comments).toMatchObject(result);
+      test("New comment is stored on the database", () => {
+        return request(app).post("/api/articles/2/comments")
+          .send({
+            username: "lurker",
+            body: "Who wrote this?"
+          }).then(() => {
+            return request(app).get("/api/articles/2/comments")
+              .then(({body}) => {
+                console.log(body.comments);
+                expect(body.comments.filter((comment) => comment.author === "lurker").length).toBe(1);
+              });
+          });
```

Comment:
Please add another test with status 201, where extra properties are ignored in the post object. For example when it looks like this: ```
{
            username: "lurker",
            body: "Who wrote this?",
            extraProp: "extraValue"
          }
```